src/server/lib/retry.ts

```diff
@@ -0,0 +1,40 @@
+interface retryParams<T> {
+  fn: () => Promise<T>;
+  retriesLeft?: number;
+  interval?: number;
+  exponential?: boolean;
+}
+
+/**
+ * Retries the given function until it succeeds given a number of retries and
+ * an interval between them. By default, will retry the function 3 times with
+ * 500ms in between attempts.
+ * Adapted from https://gitlab.com/-/snippets/1775781
+ * @param {Function} params.fn - Returns a promise
+ * @param {Number} params.retriesLeft - Number of retries. If -1 will keep retrying
+ * @param {Number} params.interval - Milliseconds between retries. If exponential is
+ * set to true, will be doubled each retry
+ * @param {Boolean} params.exponential - Flag for exponential back-off mode
+ * @return {Promise<*>}
+ */
+export default async function retry<T>({
```

Comment:
You may not need this retry method! The AWS SES library seems to have a builtin retry mechanism available as a config option!

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#maxRetries-property

Reply:
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#retryDelayOptions-property

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Request.html#retry-event